File: app/src/main/java/org/dash/dashj/demo/ui/adapter/holder/PeerViewHolder.java
package org.dash.dashj.demo.ui.adapter.holder;

import android.annotation.SuppressLint;
import android.graphics.Typeface;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import org.bitcoinj.core.Peer;
import org.bitcoinj.core.VersionMessage;
import org.dash.dashj.demo.R;

import java.net.InetAddress;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PeerViewHolder extends RecyclerView.ViewHolder {

    @BindView(R.id.peer_list_row_ip)
    TextView ipView;

    @BindView(R.id.peer_list_row_height)
    TextView heightView;

    @BindView(R.id.peer_list_row_version)
    TextView versionView;

    @BindView(R.id.peer_list_row_protocol)
    TextView protocolView;

    @BindView(R.id.peer_list_row_ping)
    TextView pingView;

    public PeerViewHolder(final View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }

    @SuppressLint("SetTextI18n")
    public void bind(Peer peer) {

        final VersionMessage versionMessage = peer.getPeerVersionMessage();
        final boolean isDownloading = peer.isDownloadData();

        final InetAddress address = peer.getAddress().getAddr();
        ipView.setText(address.getHostAddress());

        final long bestHeight = peer.getBestHeight();
        heightView.setText(bestHeight > 0 ? bestHeight + " blocks" : null);
        heightView.setTypeface(isDownloading ? Typeface.DEFAULT_BOLD : Typeface.DEFAULT);

        versionView.setText(versionMessage.subVer);
        versionView.setTypeface(isDownloading ? Typeface.DEFAULT_BOLD : Typeface.DEFAULT);

        protocolView.setText("protocol: " + versionMessage.clientVersion);
        protocolView.setTypeface(isDownloading ? Typeface.DEFAULT_BOLD : Typeface.DEFAULT);

        final long pingTime = peer.getPingTime();
        pingView.setText(pingTime < Long.MAX_VALUE ? getString(R.string.peer_list_row_ping_time, pingTime) : null);
        pingView.setTypeface(isDownloading ? Typeface.DEFAULT_BOLD : Typeface.DEFAULT);
    }

    private String getString(int resId, Object... formatArgs) {
        return itemView.getContext().getString(resId, formatArgs);
    }
}

File: app/src/main/java/org/dash/dashj/demo/ui/adapter/holder/PeerViewHolder.kt
package org.dash.dashj.demo.ui.adapter.holder

import android.graphics.Typeface
import android.support.v7.widget.RecyclerView
import android.view.View
import kotlinx.android.extensions.LayoutContainer
import kotlinx.android.synthetic.main.peer_list_row.*
import org.bitcoinj.core.Peer
import org.dash.dashj.demo.R

class PeerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), LayoutContainer {

    override val containerView: View?
        get() = itemView

    fun bind(peer: Peer) {
        val versionMessage = peer.peerVersionMessage
        val isDownloading = peer.isDownloadData

        val address = peer.address.addr
        ipView.text = address.hostAddress

        val bestHeight = peer.bestHeight
        heightView.text = if (bestHeight > 0) getString(R.string.peer_list_row_blocks, bestHeight) else null
        heightView.typeface = if (isDownloading) Typeface.DEFAULT_BOLD else Typeface.DEFAULT

        versionView.text = versionMessage.subVer
        versionView.typeface = if (isDownloading) Typeface.DEFAULT_BOLD else Typeface.DEFAULT

        protocolView.text = getString(R.string.peer_list_row_protocol, versionMessage.clientVersion)
        protocolView.typeface = if (isDownloading) Typeface.DEFAULT_BOLD else Typeface.DEFAULT

        val pingTime = peer.pingTime
        pingView.text = if (pingTime < java.lang.Long.MAX_VALUE) getString(R.string.peer_list_row_ping_time, pingTime) else null
        pingView.typeface = if (isDownloading) Typeface.DEFAULT_BOLD else Typeface.DEFAULT
    }

    private fun getString(resId: Int, vararg formatArgs: Any): String {
        return itemView.context.getString(resId, *formatArgs)
    }
}
